README.rst

```diff
@@ -20,33 +20,85 @@ configurations remain synchronized with the state of all databases in the
 cluster. PgQ is used for buffering mutation records until they are collected by
 the relay and forwarded to their final destination.
 
+Usage
+=====
+
+PGShovel is a tool which replicates mutations (``INSERT``, ``UPDATE`` and ``DELETE`` operations) made to a PostgreSQL data out of the database and into an ordered stream of transaction-aware mutation messages.  These messages can then be consumed by 3rd party systems as needed.  Commonly, the mutation stream is consumed by data warehouse operations, cache updating services, search index services or can even be processed into other new streams.
+
+At Disqus PGShovel is actively being developed to replicate changes from our master Postgres database - which contains the core commenting and other relational data - into our data warehousing system. There is further planned work to also consume the mutation stream which is models mutations to rows and columns into a new stream of **object snapshots**.  Unlike mutations, these snapshots accurately and succinctly model our business objects and are free from the idiosyncrasies of attempting to read and interpret their serialized structure in the rows and columns of an RDBMS.
+
 Concepts
 ========
 
+PGShovel has a few core concepts which are important to understanding it:
+
 Cluster
 -------
 
-A cluster contains replication sets.
+A Cluster contains a collection of Replication Sets.
 
 Replication Set
 ---------------
 
-Replication sets contain a collection of tables that are replicated together.
+Replication Sets contain configuration defining a collection of tables that are replicated together.
+
+Mutation Log
+------------
+
+The Mutation Log is where database mutation events are buffered before being forwarded by
+the Relay to their final destination. Internally, PGShovel uses the PgQ tool to queue these mutations inside the database.
+
+Batch Operations
+----------------
+
+The smallest unit of data in PGShovel.  A Batch Operation is a message describing an action describing one of 4 possible operations:
```

Comment:
Second sentence is awkward. Here are two simplifications which may be better:

>  A Batch Operation is a message describing one of 4 possible actions:

or

>  A Batch Operation is a message communicating one of 4 possible actions: